libimage/manifest_list.go

```diff
@@ -101,8 +109,153 @@ func (r *Runtime) lookupManifestList(name string) (*Image, manifests.List, error
 	return image, list, nil
 }
 
+// ConvertToManifestList converts the image into a manifest list if it is not
+// already also a list.  An error is returned if the conversion fails.
+func (i *Image) ConvertToManifestList(ctx context.Context) (*ManifestList, error) {
+	// If we don't need to do anything, don't do anything.
+	if list, err := i.ToManifestList(); err == nil {
```

Comment:
Should this be 
```
if list, err := !Errors.Is(err, ErrNotAManifestList) {
   return list, err
}
```

Reply:
Changed it to return if `err` is `nil` or something other than a `ErrNotAManifestList` error.